src/Kafka/Consumer.purs

```diff
@@ -121,18 +142,23 @@ type ConsumerConfigImpl =
 -- |   * auto commit offsets periodically during a batch
 -- |   * see [autoCommit](https://kafka.js.org/docs/consuming#a-name-auto-commit-a-autocommit)
 -- | * `eachBatch`
+-- |   * `autoResolve`
+-- |     * auto commit offsets after successful `eachBatch`
+-- |     * default: `true`
 -- |   * see [eachBatch](https://kafka.js.org/docs/consuming#a-name-each-batch-a-eachbatch)
--- | * `eachBatchAutoResolve`
--- |   * auto commit offsets after successful `eachBatch`
--- |   * default: `true`
+-- | * `partitionsConsumedConcurrently`
+-- |   * concurrently instead of sequentially invoke `eachBatch`/`eachMessage` for multiple partitions if count is greater than `1`. Messages in the same partition are still guaranteed to be processed in order, but messages from multiple partitions can be processed at the same time.
+-- |   * should not be larger than the number of partitions consumed
+-- |   * default: `1`
+-- |   * see [Partition-aware concurrency](https://kafka.js.org/docs/consuming#a-name-concurrent-processing-a-partition-aware-concurrency)
 type ConsumerRunConfig =
   { autoCommit ::
       Data.Maybe.Maybe
         { autoCommitInterval :: Data.Maybe.Maybe Number
         , autoCommitThreshold :: Data.Maybe.Maybe Number
         }
-  , eachBatch :: EachBatchHandler
-  , eachBatchAutoResolve :: Data.Maybe.Maybe Boolean
+  , consume :: Consume
+  , partitionsConsumedConcurrently :: Data.Maybe.Maybe Int
```

Comment:
❓ Should we just always be explicit about `partitionsConsumedConcurrently`. Seems like it would save us from any changes in the underlying implementation.

Reply:
I'm trying to be explicit about everything and sticking to primitive PS types in the public API. We can wrap the public API with common defaults when using it in our project.

Reply:
I'm saying `partitionsConsumedConcurrently` can just be `Int` and we can pass `1` for a single partition.

Reply:
But I can see you're trying to stick closely to the JS API which has this value as optional.

Reply:
> I'm saying `partitionsConsumedConcurrently` can just be `Int` and we can pass `1` for a single partition.

Gotcha

> But I can see you're trying to stick closely to the JS API which has this value as optional.

Yeah if it's set to `Nothing`, the default value (`1`) from `kafkajs` will be used.